truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InstrumentRegistrationProcessor.java

```diff
@@ -86,6 +96,31 @@ private void generateFile(List<TypeElement> instruments) {
         }
     }
 
+    private static int findInstrument(String id, Properties p) {
+        int cnt = 1;
+        Object val;
+        while ((val = p.get("instrument" + cnt + ".id")) != null) {
+            if (id.equals(val)) {
+                return cnt;
+            }
+            cnt += 1;
```

Comment:
You optimize for the normal Java reader and not for yourself. i'd guess that cnt++ reads more common for them. I might be wrong. Feel free to decide yourself.